src/main/scala/seek/aws/sns/SubscribeTopic.scala

```diff
@@ -26,17 +27,23 @@ class SubscribeTopic extends AwsTask {
 
   override def run: IO[Unit] =
     for {
-      t  <- topicArn.run
+      arns <- topicArns.run
+      ts = split(arns)
       c  <- buildClient(AmazonSNSClientBuilder.standard())
       ps <- gather(pending)
-      es <- listSubscriptions(t).run(c).compile.toList
-      _  <- addSubscriptions(t, uniquePending(ps, es).toList).run(c)
+      _ <- ts.foldLeft(IO.unit) { (z, t) =>
```

Comment:
Would probably read nicer if you just updated the `addSubscriptions` function to incorporate the logic of only adding non-existing subscriptions. That way your wouldn't need nested for comprehensions.